docs/lang/articles/misc/gui.md

```diff
@@ -79,11 +99,158 @@ The data type of `img` must be one of:
+
+![circles](../static/assets/colored_circles.png)
+
+[`gui.line(begin, end)`](https://api-docs.taichi.graphics/src/taichi.misc.html?highlight=line#taichi.misc.gui.GUI.line)
+draws one line.
+
+The color and radius of lines can be further specified with additional parameters.
+
+[`gui.lines(begin, end)`](https://api-docs.taichi.graphics/src/taichi.misc.html?highlight=line#taichi.misc.gui.GUI.lines)
+draws lines.
+
+:::note
+
+`being` and `end` both require input of positions.
+
+The color and radius of lines can be further specified with additional parameters.
+
+For example:
+```python
+gui.lines(begin=X, end=Y, radius=2, color=0x068587)
+```
+draws line segments from X positions to Y positions with width of 2 and color of light blue.
+
+![lines](../static/assets/lines.png)
+
+[`gui.triangle(a, b, c)`](https://api-docs.taichi.graphics/src/taichi.misc.html?highlight=triangle#taichi.misc.gui.GUI.triangle)
+draws one triangle.
+
+The color of triangles can be further specified with additional parameter.
+
+[`gui.triangles(a, b, c)`](https://api-docs.taichi.graphics/src/taichi.misc.html?highlight=triangles#taichi.misc.gui.GUI.triangles)
+draws solid triangles.
+
+Three vertices of triangles all require input of positions.
+
+The color of triangles can be further specified with additional parameter.
+
+For example:
+```python
+gui.triangles(a=X, b=Y, c=Z, color=0xED553B)
+```
+draws triangles with color of red and three points positioned at X, Y, and Z.
+
+![triangles](../static/assets/triangles.png)
+
+[`gui.rect(topleft, bottomright)`](https://api-docs.taichi.graphics/src/taichi.misc.html?highlight=rect#taichi.misc.gui.GUI.rect)
+draws a hollowed rectangle.
+
+The color and radius of the stroke of rectangle can be further specified with additional parameters.
+
+For example:
+```python
+gui.rect([0, 0], [0.5, 0.5], radius=1, color=0xED553B)
+```
+draws a rectangle of top left corner at [0, 0] and bottom right corner at [0.5, 0.5], with stroke of radius of 1 and color of red.
+
+![rect](../static/assets/rect.png)
+
+[`gui.arrows(origin, direction)`](https://api-docs.taichi.graphics/src/taichi.misc.html?highlight=arrows#taichi.misc.gui.GUI.arrows)
+draws arrows.
+
+`origin` and `direction` both require input of positions. `origin` refers to the positions of arrows' origins, `direction`
+refers to the directions where the arrows point to relative to their origins.
+
+The color and radius of arrows can be further specified with additional parameters.
+
+For example:
+```python
+x = nunpy.array([[0.1, 0.1], [0.9, 0.1]])
+y = nunpy.array([[0.3, 0.3], [-0.3, 0.3]])
+gui.arrows(x, y, radius=1, color=0xFFFFFF)
+```
+draws two arrow originated at [0.1, 0.1], [0.9, 0.1] and pointing to [0.3, 0.3], [-0.3, 0.3] with radius of 1 and color of white.
+
+![arrows](../static/assets/arrows.png)
 
+[`gui.arrow_field(direction)`](https://api-docs.taichi.graphics/src/taichi.misc.html?highlight=arrow_field#taichi.misc.gui.GUI.arrow_field)
```

Comment:
It supports both Taichi fields and numpy ndarrays right? Using the term field might confusing the readers.